code/modules/chemistry/tools/dispensers.dm

```diff
@@ -134,6 +134,19 @@
 					qdel(src)
 		return
 
+	attackby(obj/item/P, mob/user, silent = FALSE)
+		if (istype(P, /obj/item/magnifying_glass))
+			boutput(user, SPAN_NOTICE("You angle the light through the magnifying glass towards the ants."))
+			SETUP_GENERIC_ACTIONBAR(user, src, 3 SECONDS, PROC_REF(burn_ants), list(user, P), P.icon, P.icon_state, null, INTERRUPT_MOVE | INTERRUPT_STUNNED | INTERRUPT_ACT)
+		return
```

Comment:
it looks like this already has another attackby proc definition, they should be combined

```suggestion
	attackby(obj/item/I, mob/user, silent = FALSE)
		if (istype(I, /obj/item/magnifying_glass))
			boutput(user, SPAN_NOTICE("You angle the light through the magnifying glass towards the ants."))
			SETUP_GENERIC_ACTIONBAR(user, src, 3 SECONDS, PROC_REF(burn_ants), list(user, P), P.icon, P.icon_state, null, INTERRUPT_MOVE | INTERRUPT_STUNNED | INTERRUPT_ACT)
			return
		..(W, user)
		SPAWN(1 SECOND)
			if (src?.reagents)
				if (src.reagents.total_volume <= 1)
					qdel(src)
```